srtcore/buffer.cpp

```diff
@@ -1201,19 +1207,21 @@ bool CRcvBuffer::getRcvFirstMsg(steady_clock::time_point& w_tsbpdtime,
         }
         else
         {
-            /* We got the 1st valid packet */
-            w_tsbpdtime = getPktTsbPdTime(m_pUnit[i]->m_Packet.getMsgTimeStamp());
-            if (w_tsbpdtime <= steady_clock::now())
+            tsbpdtime = getPktTsbPdTime(m_pUnit[i]->m_Packet.getMsgTimeStamp());
+            if (tsbpdtime <= steady_clock::now())
             {
                 /* Packet ready to play */
+                w_tsbpdtime = tsbpdtime;
+                w_curpktseq = m_pUnit[i]->m_Packet.m_iSeqNo;
                 if (haslost)
+                    w_skipseqno = w_curpktseq;
+
+                if (base_seq != SRT_SEQNO_NONE && CSeqNo::seqoff(base_seq, w_curpktseq) <= 0)
```

Comment:
When you don't need the offset value, it is better to use `seqcmp` instead of `seqoff`.

```suggestion
                if (base_seq != SRT_SEQNO_NONE && CSeqNo::seqcmp(w_curpktseq, base_seq) <= 0)
```

Reply:
great, I will fix it later

Reply:
You can simply commit the suggestion via GitHub as an option. There is a "commit suggestion" button underneath the suggestion.
From the "Files changed" tab those two suggestions can be added to a batch commit, to commit both at once.

Reply:
thx